### ISSUE TITLE
Code Highlighter Control?

### ISSUE THREAD

**Issue by cra0zy** [picoe/Eto#422]
Is there any code editor / highlighter control for Eto Forms?

If the answer is no, I will probably end up porting https://github.com/mono/monodevelop/tree/master/main/src/core/Mono.Texteditor to Eto Forms, so do you want me to add it to this repo?

**Comment by JosefNemec** [comment]
No there isn't. I personally ended up using [Ace editor](https://ace.c9.io) via WebView control, however it has its limitations mainly because JS<->C# data exchange in Eto is not exactly flashed out at the moment, mainly when speaking about JS callbacks.

**Comment by cwensley** [comment]
Yeah @JosefNemec is right.  I would certainly have use for an editor/highlighter if there was one.  Mono.TextEditor is a good candidate, though I couldn't figure out the license as the [FAQ](http://www.monodevelop.com/help/faq/#under-which-license-is-monodevelop-available) says the core is LGPLv2, which it is under in [github](https://github.com/mono/monodevelop/tree/master/main/src/core), but the license on the source files appear to be MIT..

**Comment by cra0zy** [comment]
I am 99% certain that the License for it is MIT, reason: https://github.com/luiscubal/XwtPlus.TextEditor

Side note, how do you get Eto WebView running on Ubuntu, Gtk 3. I don't think Ubuntu has any official packages of webkit-sharp for Gtk 3.

**Comment by cwensley** [comment]
@cra0zy yeah there's no webkit-sharp package for Gtk 3.  I believe I've compiled it from source and got it working at one point though..

**Comment by SlowLogicBoy** [comment]
> I personally ended up using Ace editor via WebView control, however it has its limitations mainly because JS<->C# data exchange in Eto is not exactly flashed out at the moment, mainly when speaking about JS callbacks.

Yesterday I took a bit different approach:
I created service with HttpListener (although better and easier solution can be used), loaded Ace and navigated WebView to it. Then I could handle event's through that HttpListener (nothing through WebView).

It's a quite possible and good work around. however, you will need to develop a webpage which I personally hate.

So I'm just bumping this issue up, to hopefully make this a bit higher priority :)
